frontend/components/bc/BcInputSearch.vue

```diff
@@ -0,0 +1,322 @@
+<script setup lang="ts" generic="T">
+const props = withDefaults(defineProps<{
+  hasError?: boolean,
+  hasFocus?: boolean,
+  isLoading: boolean,
+  results: T[] | undefined,
+  shouldClearOnSubmit?: boolean,
+  shouldSelectFirstResult?: boolean,
+}>(), {
+  shouldClearOnSubmit: true,
+})
+
+const emit = defineEmits<{
+  (e: 'search', value: string): void,
+  (e: 'submit', value: T): void,
+}>()
+
+const { t: $t } = useTranslation()
+
+const input = defineModel<string>()
+
+const elementInput = ref<HTMLElement >()
```

Comment:
np: (nvm I will research ESLint rule for that) 
```suggestion
const elementInput = ref<HTMLElement>()
```

Reply:
kwl thx...so I leave it in for you to see how if it works